File: MobileApp/hirafi_frontend/lib/models/work_request.dart
class WorkRequest {
  final int id;
  final int userId;
  final String userFirstName;
  final String userLastName;
  final int workerId;
  final String workerFirstName;
  final String workerLastName;
  final String workerProfession;
  final String description;
  final String? preferredDate;
  final String? location;
  final String status;
  final DateTime createdAt;

  WorkRequest({
    required this.id,
    required this.userId,
    required this.userFirstName,
    required this.userLastName,
    required this.workerId,
    required this.workerFirstName,
    required this.workerLastName,
    required this.workerProfession,
    required this.description,
    this.preferredDate,
    this.location,
    required this.status,
    required this.createdAt,
  });

  factory WorkRequest.fromJson(Map<String, dynamic> json) {
    return WorkRequest(
      id: json['id'] ?? 0,
      userId: json['userId'] ?? 0,
      userFirstName: json['userFirstName'] ?? '',
      userLastName: json['userLastName'] ?? '',
      workerId: json['workerId'] ?? 0,
      workerFirstName: json['workerFirstName'] ?? '',
      workerLastName: json['workerLastName'] ?? '',
      workerProfession: json['workerProfession'] ?? '',
      description: json['description'] ?? '',
      preferredDate: json['preferredDate'],
      location: json['location'],
      status: json['status'] ?? 'PENDING',
      createdAt: DateTime.tryParse(json['createdAt'] ?? '') ?? DateTime.now(),
    );
  }
}

File: MobileApp/hirafi-backend/src/main/java/com/hirafi/backend/entity/WorkRequest.java
package com.hirafi.backend.entity;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@Table(name = "work_requests")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Builder
public class WorkRequest {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "worker_id", nullable = false)
    private Worker worker;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "preferred_date")
    private LocalDate preferredDate;

    private String location;

    @Column(precision = 10, scale = 2)
    private java.math.BigDecimal amount;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private RequestStatus status = RequestStatus.PENDING;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}
